File: src/main/resources/static/script/ProfileIntention.ts
export enum ProfileIntention {
    //============================-Variables-=================================
    NETWORKING = 'Networking',
    GETTING_INVOLVED = 'Getting Involved',
    CONNECT_TEAM = 'Connect with Team',
    FIND_TEAM = 'Find a Team',
    SOCIALIZE = 'Socialize',
    JOB_SEARCH = 'Job Search',
    LEARN = 'Learn'
}
export namespace ProfileIntention {
    export function from(profileIntentionString: string): ProfileIntention {
        let profileIntention: ProfileIntention;
        switch(profileIntentionString) {
            case ProfileIntention.NETWORKING:
                profileIntention = ProfileIntention.NETWORKING;
                break;
            case ProfileIntention.GETTING_INVOLVED:
                profileIntention = ProfileIntention.GETTING_INVOLVED;
                break;
            case ProfileIntention.CONNECT_TEAM:
                profileIntention = ProfileIntention.CONNECT_TEAM;
                break;
            case ProfileIntention.FIND_TEAM:
                profileIntention = ProfileIntention.FIND_TEAM;
                break;
            case ProfileIntention.SOCIALIZE:
                profileIntention = ProfileIntention.SOCIALIZE;
                break;
            default:
                throw new Error('Invalid Profile Intention');
        }
        return profileIntention;
    }
}

File: src/main/java/org/theoliverlear/entity/user/personal/ProfileIntention.java
package org.theoliverlear.entity.user.personal;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.Transient;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Embeddable
public class ProfileIntention {
    @Column(name = "profile_intention")
    private String profileIntention;
    @Transient
    @JsonIgnore
    public static final ProfileIntention NETWORKING = new ProfileIntention("Networking");
    @Transient
    @JsonIgnore
    public static final ProfileIntention GETTING_INVOLVED = new ProfileIntention("Getting Involved");
    @Transient
    @JsonIgnore
    public static final ProfileIntention CONNECT_TEAM = new ProfileIntention("Connect with Team");
    @Transient
    @JsonIgnore
    public static final ProfileIntention FIND_TEAM = new ProfileIntention("Find a Team");
    @Transient
    @JsonIgnore
    public static final ProfileIntention SOCIALIZE = new ProfileIntention("Socialize");
    @Transient
    @JsonIgnore
    public static final ProfileIntention JOB_SEARCH = new ProfileIntention("Job Search");
    @Transient
    @JsonIgnore
    public static final ProfileIntention LEARN = new ProfileIntention("Learn");
    public ProfileIntention() {
        this.profileIntention = ProfileIntention.SOCIALIZE.profileIntention;
    }
    public ProfileIntention(String profileIntention) {
        this.profileIntention = profileIntention;
    }
    public static ProfileIntention from(String profileIntention) {
        return switch (profileIntention) {
            case "Networking" -> ProfileIntention.NETWORKING;
            case "Getting Involved" -> ProfileIntention.GETTING_INVOLVED;
            case "Connect with Team" -> ProfileIntention.CONNECT_TEAM;
            case "Find a Team" -> ProfileIntention.FIND_TEAM;
            case "Socialize" -> ProfileIntention.SOCIALIZE;
            case "Job Search" -> ProfileIntention.JOB_SEARCH;
            case "Learn" -> ProfileIntention.LEARN;
            default -> new ProfileIntention(profileIntention);
        };
    }
    public static ProfileIntention from(ProfileIntention profileIntention) {
        return switch (profileIntention.getProfileIntention()) {
            case "Networking" -> ProfileIntention.NETWORKING;
            case "Getting Involved" -> ProfileIntention.GETTING_INVOLVED;
            case "Connect with Team" -> ProfileIntention.CONNECT_TEAM;
            case "Find a Team" -> ProfileIntention.FIND_TEAM;
            case "Socialize" -> ProfileIntention.SOCIALIZE;
            case "Job Search" -> ProfileIntention.JOB_SEARCH;
            case "Learn" -> ProfileIntention.LEARN;
            default -> new ProfileIntention(profileIntention.getProfileIntention());
        };
    }
}
